apps/ledger-live-mobile/src/newArch/features/LandingPages/screens/LargeMoverLandingPage/index.tsx

```diff
@@ -27,43 +50,49 @@ export const LargeMoverLandingPage = ({ route }: LargeMoverLandingPageProps) =>
 
   const [currentIndex, setCurrentIndex] = useState(0);
 
-  const handleNext = () => {
-    setCurrentIndex(prevIndex => (prevIndex + 1) % currencies.length);
-  };
+  const navigation = useNavigation<NavigationProp<WalletTabNavigatorStackParamList>>();
 
-  if (loading) {
-    return (
-      <Flex height={height} width="100%" justifyContent="center" alignItems="center">
-        <InfiniteLoader color="primary.c50" size={38} />
-      </Flex>
-    );
-  }
+  const currenciesWithId = useMemo(
+    () =>
+      currencies.map((currency, index) => ({
+        ...currency,
+        idCard: index,
+        range,
+        setRange,
+      })),
+    [currencies, range, setRange],
+  );
+
+  const renderCard = (card: Card) => {
+    if (!card.data || !card.chartData) {
+      return null;
+    }
+    return <Card {...card.data} chartData={card.chartData} range={range} setRange={setRange} />;
+  };
 
   if (currencies.length === 0) {
```

Comment:
How can it be equal to 0 ? 

It will always be defined even if all queries are in error no? I think it's not the good assertion here